lib/matplotlib/axis.py

```diff
@@ -1833,14 +1833,28 @@ def set_ticks(self, ticks, *, minor=False):
             self.set_major_locator(mticker.FixedLocator(ticks))
             return self.get_major_ticks(len(ticks))
 
-    def _get_tick_boxes_siblings(self, xdir, renderer):
+    def _get_tick_boxes_siblings(self, renderer):
         """
         Get the bounding boxes for this `.axis` and its siblings
-        as set by `.Figure.align_xlabels` or  `.Figure.align_ylablels`.
+        as set by `.Figure.align_xlabels` or  `.Figure.align_ylabels`.
 
         By default it just gets bboxes for self.
         """
-        raise NotImplementedError('Derived must override')
+        bboxes = []
+        bboxes2 = []
+        # Get the Grouper keeping track of x or y label groups for this figure.
+        axis_name, = [
+            name for name, axis in self.axes._get_axis_map().items()
+            if name in self.figure._align_label_groups and axis is self]
```

Comment:
I don't claim to fully understand this. But are we sure the comprehension always returns exactly one element? Particular also for derived classes like `ThetaAxis`?

Otherwise, it may be safer to catch that and `return [], []`, maybe with a warning. I assume it's better in such a case to not consider siblings than to raise.

Reply:
`polar` is fine (as _get_axis_map still knows the axises as 'x' and 'y' (this is documented in the _get_axis_map docstring)), but I agree better being safe than sorry.